packages/web3-providers/src/treasure/index.ts

```diff
@@ -0,0 +1,358 @@
+                            address: event.buyer.address || '',
+                            profile_img_url: '',
+                            link: '',
+                        },
+                    }
+                }
+                return
+            },
+        )
+
+        return offers.filter((offer) => offer !== undefined)
+    }
+
+    async getListings(tokenAddress: string, tokenId: string): Promise<NonFungibleTokenAPI.AssetOrder[]> {
+        const variables = {
+            tokenAddress,
+            tokenId,
+        }
+
+        const listingTreasure = await this.client.request(getTokenDetails, variables)
+
+        const orders: NonFungibleTokenAPI.AssetOrder[] = listingTreasure.Token[0].transferEvents.map(
+            (event: Listing): NonFungibleTokenAPI.AssetOrder | undefined => {
+                if (event.status !== 1) {
+                    return {
+                        created_time: new Date(`${event.blockTimestamp}Z`).getTime().toString(),
+                        approved_on_chain: false,
+                        current_price: formatWeiToEther(event.pricePerItem || 0).toString(),
+                        payment_token: 'MAGIC',
+                        payment_token_contract: {
+                            name: 'Magic',
+                            symbol: 'MAGIC',
+                            decimals: 18,
+                            address: '0x539bdE0d7Dbd336b79148AA742883198BBF60342',
+                        },
+                        listing_time: 0,
+                        side: NonFungibleTokenAPI.OrderSide.Sell,
+                        quantity: '1',
+                        expiration_time: 0,
+                        order_hash: event.transactionLink || '',
+                        maker_account: {
+                            user: { username: event.seller.address || '' },
+                            address: event.seller.address || '',
+                            profile_img_url: '',
+                            link: '',
+                        },
+                    }
+                }
+                return
+            },
+        )
+
+        return orders.filter((order) => order !== undefined)
+    }
+
+    async getOrders(tokenAddress: string, tokenId: string, side: NonFungibleTokenAPI.OrderSide) {
+        switch (side) {
+            case NonFungibleTokenAPI.OrderSide.Buy:
+                return this.getOffers(tokenAddress, tokenId)
+            case NonFungibleTokenAPI.OrderSide.Sell:
+                return this.getListings(tokenAddress, tokenId)
+            default:
+                return []
+        }
+    }
+}
+
+export function getTreasureNFTList() {
+    return
+}
+export function getTreasureCollectionList() {
+    return
+}
+
+function convertTreasureId(address: string, tokenId: string) {
+    const hexNumber = '0x' + new BigNumber(tokenId).toString(16)
+    console.log([address, hexNumber].join('-'))
+    return [address, hexNumber].join('-')
```

Comment:
Does it necessary to checksum the address?

Reply:
good idea, i will add